### DISCUSSION TITLE
Maintenance/Security Automation (ROBOTS)

### DISCUSSION THREAD

**Discussion by PrincessMuu-Muu** [FakeFishGames/Barotrauma#5719] [Ideas and suggestions]
With all the problems associated with the fragile human crewmates, I was thinking robots might make for a good change. The idea would be adding crew who are immune to drowning, immunity/resistance to depth, immunity/resistance to fire, and "invisibility" to certain enemies. These droids might run on several batteries, or perhaps even a fuel rod. Another aspect of this idea would perhaps be adding an "easy mode" to the campaign that replaces crew with robots. Personally I want crew who aren't fragile, especially online. Players kill the current NPCs way too fast and easily. A small format change might reduce this.

Here are some of my ideas:

1. Crew droid. This droid would work exactly like normal crew members but are immune to drowning/pressure. They would be targeted by all fish and die just as easily. (Their injuries could even work the same way, oil instead of blood. Perhaps they just run out of HP and you have to spend 1 repair kit to revive them.)

2. Maintenance droid. This droid would be limited strictly to repairs and would have no combat functionality. Some variety of fish would ignore these. Another thought I had for these were maybe they could be an expensive "upgrade" where you place it on a specific system and it'll repair it as long as it's alive, basically keeping a specific device immune to breaking.

3. Combat droid. This droid would be superior in firepower and HP to human security. They wouldn't be able to repair or man anything, but they would patrol the sub and purge intruders. (You would probably have to feed these robots coilgun ammo or whatever new weapons are added for the submarine weapon update!)

4. Cargo droid. This droid is just a mule you command to follow you for mining/shipwrecks. It would severely reduce the amount of time/trips required to empty loot onto your ship. (You guys could combine this with the cargo update, giving cargo droids a mobile mass storage.)

5. Sentry turret. This is something I've wanted FOREVER. In the meantime I keep putting coilguns INSIDE my subs for emergency/lockdown situations. It would be a million times more convenient (and balanced) if it was just an NPC entity that could be destroyed with returning fire.

**Comment by Cogs16** [comment]
I agree, and I think it would be a great way to use the AI that the team has worked hard on. I think it could also simplify some of the difficulties with programming on the dev end, and also make players enjoy the NPCs more. Right now, I think a lot of players dislike bots because they see them as basically being bad replacement players. I think if there were some that appeared as androids, people will see them more as "bonus equipment" which helps them and may also see them more positively. I think if players also had to customize/interact with the androids there would be more attachment to them. 

In terms of the androids themselves, it might be neat to have a body tier system and a modular AI system. Basically there could be three chassis A level 1, a level 2, and a level 3. Level 1s look like cheap knock off humans, level 3s are indistinguishable from huimans and are essentially the same as an NPC bot as of now. The AI system would function as described above, except there would be a few "program chips" one for each class, which gives them the functionality of that class. Chassis also have more chip slots at the level goes up. A level 1 for instance could only have 1 chip, and thus would only function as one class. For instance, if the chassis had an engineer chip, it could only operate the reactor, repair electrical devices, and maybe put out fires. A level 2 chassis could have two or three, and a level 3 could either come with all preinstalled or would have 4-5 chip slots. Level ones would be cheaper than hiring crew members, but you might need more of them. There could also be personality chips, so the bots would be capable of different conversations. None of the chassis would require oxygen, however they would still be vulnerable to pressure. Tier chassis would also be much more fragile, and would have half the health of a human. Additionally, lower tier machines such as tier 1s have a small chance of being stunned by contact with water, but they could recover quickly. All NPCs would self refuel with batteries.

Lastly, there could be a traitor AI chip which basically once inserted into the bot would turn it into a griefer.

**Reply by Cogs16** [reply]
Also a way to integrate them with circuit systems would be awsome. For instance, if a wireless component outputs a signal of 1, that could cause the robot to switch its task to circuit repair.
